### PR TITLE
Modularized train.py, modified mcts to fit with it

### LINKED ISSUE #120: Refactor train.py to allow stockfish-train implementation and self-play to use common subclass
I may have not understood the implementation idea entirely, but from @fshcat:
> alternatively maybe have stockfish and mcts both inherit some class where one of the functions takes a board state and returns probabilities and a value
> then train.py just needs to take that superclass as a parameter and call that one function
> so the game loop can still be in train.py while still having it be able to take both

This will allow adding in code to bootstrap the training by learning initial values for the neural network from output of stockfish evaluation.

### LINKED ISSUE #121: Add implementation of Stockfish-based training to bootstrap network weight initialization
This is dependent on completing #120. A new file will be created to interact with `train.py` to allow training based on the output of stockfish position evaluation.

### PR DESCRIPTION
**Pull request by fshcat**
(resolves #120)

### CODE REVIEW

**Review by ryPattillo — COMMENTED** [review]
Looks good to me

File: chess-AI/mcts.py [hunk 1]
```diff
@@ -42,6 +42,14 @@ def get_best_move(self, mcts_simulations, board, nnet):
...
         return best_move
 
+    def get_tree_results(self, mcts_simulations, nnet, board, temperature):
+        for _ in range(mcts_simulations):
+            self.search(board, nnet)
+
+        return self.find_search_probs(board.fen(), temperature)
+
+
+
```
Comment (nashirj) [inline chess-AI/mcts.py:52]: Only one space between functions for consistency pls

File: chess-AI/train.py [hunk 2]
```diff
@@ -22,19 +22,23 @@ class MctsTrain:
...
-        mcts_probs = []
+        # board -> move_list, move_probabilities, move_to_take
+        self.move_approximator = move_approximator
+
+        # board -> state_value
+        # if None, values will be based on the game outcome
+        self.val_approximator = val_approximator
+
+    def training_game(self):
+        # Stores probability distributions and values from the approximators
```
Comment (nashirj) [inline chess-AI/train.py:39]: This comment should 1) be a docstring and 2) explain the return values more clearly

File: chess-AI/train.py [hunk 3]
```diff
@@ -148,8 +153,14 @@ def training_episode(self, nnet, games, epochs, batch_size):
 
 def main():
     # Gets the neural network, and performs episodes
+    mcts_simulations = 3
+    mcts = Mcts(exploration=5)
     nnet = PlayNetwork()
-    train = MctsTrain(mcts_simulations=3, exploration=5, lr=0.1)
+
+    # def mcts_moves(board): mcts.get_tree_results(mcts_simulations, nnet, board, temperature=5)
```
Comment (nashirj) [inline chess-AI/train.py:160]: Can remove this line

File: chess-AI/mcts.py [hunk 4]
```diff
@@ -42,6 +42,14 @@ def get_best_move(self, mcts_simulations, board, nnet):
 
         return best_move
 
+    def get_tree_results(self, mcts_simulations, nnet, board, temperature):
```
Comment (nashirj) [inline chess-AI/mcts.py:45]: Please add a docstring

**Review by nashirj — COMMENTED** [review]
I know this is merged already but can you pls make the changes I suggested in a new PR

**Pull request merged**